### PR TITLE
Add bandmeter buckettoken

### PR DESCRIPTION
**Pull request by qiaozhigang**

### CODE REVIEW

File: p2p/fechelper.go [hunk 1]
```diff
@@ -1,2133 +1,98 @@
...
-	"github.com/aristanetworks/goarista/monotime"
-)
-
-//CFECHelper helper class for fec
-type CFECHelper struct {
+//FECHelper helper class for FEC
+type FECHelper struct {
 	canVec         [255]*VBitVec
 	bitLen, fecLen uint
+	idx            uint
```
Comment (lucusfly) [inline p2p/fechelper.go]: add comments for these fields

File: p2p/fechelper.go [hunk 2]
```diff
@@ -1,2133 +1,98 @@
...
-	p.BitXor(h.canVec[0], h.canVec[3])
-	p.BitXor1(h.canVec[4])
-	p.BitXor1(h.canVec[5])
-	p.BitXor1(h.canVec[6])
-	p.BitXor1(h.canVec[7])
-	p.ExtFlag = 0x9F
-	h.canVec[idx] = p
-	idx++
+	fmt.Print(arr)
+	fmt.Printf(",%d, %02X\r\n", arrLen, flag)
```
Comment (lucusfly) [inline p2p/fechelper.go]: remove?

File: p2p/fechelper.go [hunk 3]
```diff
@@ -1,2133 +1,98 @@
...
-	p.Init(_bitLen)
-	p.BitXor(h.canVec[0], h.canVec[2])
-	p.BitXor1(h.canVec[4])
-	p.BitXor1(h.canVec[5])
-	p.BitXor1(h.canVec[6])
-	p.BitXor1(h.canVec[7])
-	p.ExtFlag = 0xAF
-	h.canVec[idx] = p
-	idx++
+func (h *FECHelper) createOne(arr []int, arrLen int) {
```
Comment (lucusfly) [inline p2p/fechelper.go]: what don't this method do? add comments

File: p2p/fechelper_test.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+*  @copyright defined in go-seele/LICENSE
+ */
+package p2p
+
+import (
+	"fmt"
+	"testing"
+)
+
+func Test_abc3(t *testing.T) {
```
Comment (lucusfly) [inline p2p/fechelper_test.go]: refine Test_abc3

File: p2p/speedmeter.go [hunk 5]
```diff
@@ -1,26 +1,81 @@
+/**
+*  @file
+*  @copyright defined in go-seele/LICENSE
```
Comment (lucusfly) [inline p2p/speedmeter.go]: remove all these headers

File: p2p/speedmeter.go [hunk 6]
```diff
@@ -1,26 +1,81 @@
...
 //SpeedMeter compute bandwidth
 type SpeedMeter struct {
 	itemArr     []speedMeterSubItem
-	preFeedTick uint32
-	step        uint32
-	itemsNum    uint32
+	preFeedTick uint64
+	step        uint64
+	itemsNum    uint
+	mutex       sync.Mutex
```
Comment (lucusfly) [inline p2p/speedmeter.go]: add comments for fields

File: p2p/fechelper.go [hunk 7]
```diff
@@ -1,2133 +1,98 @@
...
+	}
+	h.bitLen, h.fecLen = _bitLen, 8
+
+	h.createAndInsertByPattern(0x80, 1, 0x80)
+	h.createAndInsertByPattern(0x80, 2, 0x40)
+	h.createAndInsertByPattern(0x80, 3, 0x20)
+	h.createAndInsertByPattern(0x40, 3, 0x10)
+	h.createAndInsertByPattern(0xdb, 7, 0x08)
+	h.createAndInsertByPattern(0xba, 7, 0x04)
+	h.createAndInsertByPattern(0x6e, 7, 0x02)
```
Comment (lucusfly) [inline p2p/fechelper.go]: nice!

File: p2p/speedmeter_test.go [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/aristanetworks/goarista/monotime"
+)
+
+func Test_test1second(t *testing.T) {
```
Comment (lucusfly) [inline p2p/speedmeter_test.go]: bad method name

File: p2p/speedmeter_test.go [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+	t1 := monotime.Now() / 1000
+	fmt.Println("tick1=", t1)
+	time.Sleep(1000)
+	t2 := monotime.Now() / 1000
+	fmt.Println("tick2=", t2, t2-t1)
+	sp := NewSpeedMeter(100, 10)
+	sp.Feed(128 * 1024)
+	/*var i1 int64 = 10
```
Comment (lucusfly) [inline p2p/speedmeter_test.go]: better add some check instead of print

File: p2p/tokenbucket.go [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+//TokenBucket for bucket limit rate
+type TokenBucket struct {
+	factor         float64
+	bytesPerSecond int64
+	maxTokens      int64
+	curTokens      int64
+	tokensPerMS    int64
+	reservedTokens int64
+	preTick        uint64
+	mutex          sync.Mutex
```
Comment (lucusfly) [inline p2p/tokenbucket.go]: comments for field

File: p2p/tokenbucket_test.go [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+package p2p
+
+import (
+	"fmt"
+	"testing"
+)
+
+func Test_test1(t *testing.T) {
```
Comment (lucusfly) [inline p2p/tokenbucket_test.go]: bad method name

File: .vscode/settings.json [hunk 12]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (lucusfly) [inline .vscode/settings.json]: remove vscode config file

**Comment by qiaozhigang** [comment]
ok, commit already

**Pull request merged**